guides/source/routing.md

```diff
@@ -319,19 +339,21 @@ resources :publishers do
 end
 ```
 
-Deeply-nested resources quickly become cumbersome. In this case, for example, the application would recognize paths such as:
+Deeply nested resources can become cumbersome to maintain and reason about. In the above example, the application would recognize paths such as:
 
 ```
 /publishers/1/magazines/2/photos/3
 ```
 
-The corresponding route helper would be `publisher_magazine_photo_url`, requiring you to specify objects at all three levels. Indeed, this situation is confusing enough that a [popular article by Jamis Buck](http://weblog.jamisbuck.org/2007/2/5/nesting-resources) proposes a rule of thumb for good Rails design:
-
-TIP: Resources should never be nested more than 1 level deep.
+The corresponding route helper would be `publisher_magazine_photo_url`, requiring you to specify objects at all three levels. This can become confusing and overly complex. Therefore, the general rule of thumb is to only nest resources 1 level deep.
 
 #### Shallow Nesting
 
-One way to avoid deep nesting (as recommended above) is to generate the collection actions scoped under the parent, so as to get a sense of the hierarchy, but to not nest the member actions. In other words, to only build routes with the minimal amount of information to uniquely identify the resource, like this:
+One way to avoid deep nesting (as recommended above) is to generate the
+collection actions scoped under the parent, so as to get a sense of the
+hierarchy, but to not nest the member actions. In other words, to only build
```

Comment:
Maybe it would be helpful to explicitly list out which CRUD actions (i.e. member actions) that should not be nested for beginner developers that are not quite familiar with the collection vs member terms.

Reply:
Added definition for "member" and "collection" action with example CRUD actions.